tasks/main.yml

```diff
@@ -1 +1,60 @@
 # role tasks
+
+- name: Check localectl status before
+  command: localectl status
+  register: localectl
+  changed_when: "localectl.rc != 0"
+
+- name: set the localectl before fact
+  set_fact:
+    localectl_before: "{{ localectl.stdout_lines | join ('\n') }}"
+
+- name: Set the language locale
+  command: localectl set-locale "{{ localectl_system_locale}}"
+  register: locale
+  changed_when: "locale.rc != 0"
```

Comment:
If the rc is non-zero, ansible will fail this task and the entire playbook that contains it.

This is an inappropriate use of changed_when. This command has an rc of 0 when setting the locale to its current value or to a different value; presumably an `rc !=0` means that set-locale *failed*, not that set-locale did not make a change. The output of `localectl status` should probably be used in a `when` clause for this task.

This applies to all following `set-*` tasks that use `changed_when` similarly.

Reply:
If localectl gives output, you should search the value of `locale.stdout` for the appropriate cues. Otherwise, you should guard this command and have it only run when it detects a change is needed and set `changed_when: true` here.

Basically, this should definitely have a module written, or it needs guarding around these commands.